lib/payment_processor/clients/braintree/transaction.rb

```diff
@@ -30,19 +30,8 @@ def sale
               email: @user[:email]
             }
           )
-          process_response(transaction)
         end
-
-        def process_response(braintree_response)
-          puts "processing response"
-          if braintree_response.success?
-            { success: true, transaction_id: braintree_response.transaction.id }.to_json
-          else
-            { success: false, errors: [] }.to_json
-          end
-        end
-
       end
     end
   end
```

Comment:
@Tuuleh I'd suggest you don't process the response here. The response hash is actually a view, and should be handled as a separate concern. Here, I've moved it back to the controller... for more complicated JSON objects I'd consider using `jbuilder` view templates. And if your template(s) end up having lots of view logic, then I'd use Rails' helper methods or you can wrap your transaction object with an object concerned only with presentation (https://github.com/drapergem/draper might be helpful, although probably not necessary).